### PR TITLE
[Dygraph] train and eval yolov3 successfully

### PR DESCRIPTION
**Pull request by wangxinxin08**

### CODE REVIEW

File: ppdet/modeling/backbone/darknet.py [hunk 1]
```diff
@@ -30,7 +31,12 @@ def __init__(self,
             weight_attr=ParamAttr(name=name + '.conv.weights'),
             bias_attr=False)
         bn_name = name + '.bn'
-        self.batch_norm = nn.BatchNorm2D(
+        if norm_type == 'sync_bn':
```
Comment (jerrywgz) [inline ppdet/modeling/backbone/darknet.py:34]: bn的封装放到ops里吧，这样其他模型也可以用

File: ppdet/data/reader.py [hunk 2]
```diff
@@ -11,25 +11,36 @@
...
             for k, v in t.items():
                 op_cls = getattr(from_, k)
                 self.transforms_cls.append(op_cls(**v))
                 if hasattr(op_cls, 'num_classes'):
                     op_cls.num_classes = num_classes
 
+                if op_cls in [
+                        transform.Gt2YoloTargetOp, transform.Gt2YoloTarget
+                ]:
+                    output_fields = ['image', 'gt_bbox']
```
Comment (heavengate) [inline ppdet/data/reader.py:36]: 这里的这个和fields配置不一致么

File: ppdet/data/transform/operator.py [hunk 3]
```diff
@@ -182,6 +183,56 @@ def apply(self, sample, context=None):
         return sample
 
 
+@register_op
+class RandomErasingImageOp(BaseOperator):
+    def __init__(self, prob=0.5, lower=0.02, heigher=0.4, aspect_ratio=0.3):
```
Comment (heavengate) [inline ppdet/data/transform/operator.py:188]: heigher -> higher ?

File: tools/train.py [hunk 4]
```diff
@@ -137,19 +135,28 @@ def run(FLAGS, cfg, place):
                              cfg.get('load_static_weights', False),
                              FLAGS.weight_type)
 
+    if getattr(model.backbone, 'norm_type', None) == 'sync_bn':
+        assert cfg.use_gpu and ParallelEnv(
+        ).nranks > 1, 'you should use bn rather than sync_bn while using a single gpu'
+    # sync_bn = (getattr(model.backbone, 'norm_type', None) == 'sync_bn' and
+    #            cfg.use_gpu and ParallelEnv().nranks > 1)
+    # if sync_bn:
+    #     model = paddle.nn.SyncBatchNorm.convert_sync_batchnorm(model)
```
Comment (heavengate) [inline tools/train.py:144]: remove？

File: tools/train.py [hunk 5]
```diff
@@ -137,19 +135,28 @@ def run(FLAGS, cfg, place):
...
     # Run Train
-    start_iter = 0
     time_stat = deque(maxlen=cfg.log_iter)
     start_time = time.time()
     end_time = time.time()
     # Run Train
     start_epoch = optimizer.state_dict()['LR_Scheduler']['last_epoch']
     for e_id in range(int(cfg.epoch)):
         cur_eid = e_id + start_epoch
+        train_loader.dataset.epoch = e_id
```
Comment (heavengate) [inline tools/train.py:159]: e_id -> epoch_id

File: ppdet/modeling/backbone/darknet.py (hunk 1 above)
Comment (wangxinxin08) [reply ppdet/modeling/backbone/darknet.py:34]: Done, thanks!

File: ppdet/data/transform/operator.py (hunk 3 above)
Comment (wangxinxin08) [reply ppdet/data/transform/operator.py:188]: Done, thanks!

File: tools/train.py (hunk 4 above)
Comment (wangxinxin08) [reply tools/train.py:144]: will use in the future after convert_sync_batchnorm fixed

File: tools/train.py (hunk 5 above)
Comment (wangxinxin08) [reply tools/train.py:159]: Done, thanks!

**Review by heavengate — APPROVED** [review]

**Review by yghstill — APPROVED** [review]

**Pull request merged**